components/common/typescript.gni

```diff
@@ -152,4 +152,31 @@ template("pack_web_resources") {
                             "visibility",
                           ])
   }
-}
\ No newline at end of file
+}
+
+# Runs a script to compile a local Cargo package and all of its dependencies
+template("cargo_build") {
+  action(target_name) {
+    script = "//brave/script/cargo.py"
+
+    # where to put output files
+    if (defined(invoker.output_dir)) {
+      output_dir = invoker.output_dir
+    } else {
+      cargo_name = invoker.cargo_name
+      output_dir = "$root_gen_dir/brave/components/$cargo_name/out"
```

Comment:
just small issues, but I don't think it makes sense to hard-code `brave/components` here and this should go in its own gni file

Reply:
👍